src/stores/WalletHistoryStore.js

```diff
@@ -0,0 +1,108 @@
+import { observable, action, reaction, runInAction } from 'mobx';
+import _ from 'lodash';
+
+import Transaction from './models/Transaction';
+import { queryAllTransactions } from '../network/graphQuery';
+import { SortBy, TransactionType } from '../constants';
+
+
+const INIT_VALUES = {
+  fullList: [],
+  list: [],
+  orderBy: 'createdTime',
+  direction: SortBy.Descending,
+  limit: 500,
+  skip: 0,
+  perPage: 10,
+  page: 0,
+};
+
+export default class WalletHistoryStore {
+  @observable fullList = INIT_VALUES.fullList
+  @observable list = INIT_VALUES.list
+  @observable orderBy = INIT_VALUES.orderBy
+  @observable direction = INIT_VALUES.direction
+  @observable limit = INIT_VALUES.limit
+  @observable skip = INIT_VALUES.skip
+  @observable perPage = INIT_VALUES.perPage
+  @observable page = INIT_VALUES.page
+
+  constructor() {
+    reaction(
+      () => this.page,
+      () => {
+        const start = this.page * this.perPage;
+        this.list = _.slice(this.fullList, start, start + this.perPage);
+      }
+    );
+
+    reaction(
+      () => this.perPage,
+      () => this.list = _.slice(this.fullList, 0, this.perPage),
+    );
+
+    reaction(
+      () => this.orderBy,
+      () => this.list = _.orderBy(this.list, [this.orderBy], [this.direction]),
+    );
+
+    reaction(
+      () => this.direction,
+      () => this.list = _.orderBy(this.list, [this.orderBy], [this.direction]),
+    );
+  }
+
+  @action
+  init = async () => {
+    this.reset();
+    await this.queryTransactions();
+  }
+
+  @action
+  queryTransactions = async (orderBy = this.orderBy, direction = this.direction, limit = this.limit, skip = this.skip) => {
+    try {
+      const filters = [{ type: TransactionType.Transfer }];
+      const orderByObj = { field: orderBy, direction };
+      const result = await queryAllTransactions(filters, orderByObj, limit, skip);
+
+      runInAction(() => {
+        this.fullList = _.map(result, (tx) => new Transaction(tx));
+        this.list = _.slice(this.fullList, 0, this.perPage);
+      });
+    } catch (error) {
+      console.error(error); // eslint-disable-line
+
+      runInAction(() => {
+        this.fullList = [];
+        this.list = [];
+      });
+    }
+  }
+
+  @action
+  onPageChange = (page) => {
```

Comment:
so, I would prefer we not have `setter` methods like this, but if we're hard on it then it's not the end of the world. I think it just removes redundant code and increases readability of the tests. For instance
```js
describe('test something', async () => {
  const appStore = new AppStore()
  const { walletHistory } = appStore
  await walletHistory.init()
  walletHistory.page = 1 // <- setting this way I think is more clear/easier to read than
  walletHistory.onPageChange(1) // as opposed to this
  expect(walletHistory.list).toEqual(...)
})

```

Reply:
oh i see. i'm ok with not having setter methods too. didn't know you wanted to just set everything directly from the comp. no problem with me, fixed